nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf

```diff
@@ -37,21 +39,33 @@ if (params.input) { ch_input = file(params.input, checkIfExists: true) } else {
  * Reference genomes
  */
 // TODO nf-core: Add any reference files that are needed
-// NOTE - FOR SIMPLICITY THIS IS NOT USED IN THIS PIPELINE, EXAMPLE ONLY
-// If you want to use the channel below in a process, define the following:
-//   input:
-//   file fasta from ch_fasta
-//
+// NOTE - FOR SIMPLICITY THIS IS NOT USED IN THIS PIPELINE
+// EXAMPLE ONLY TO DEMONSTRATE USAGE OF AWS IGENOMES
+if (params.genomes && params.genome && !params.genomes.containsKey(params.genome)) {
+    exit 1, "The provided genome '${params.genome}' is not available in the iGenomes file. Currently the available genomes are ${params.genomes.keySet().join(", ")}"
+}
 params.fasta = params.genomes[params.genome]?.fasta
 if (params.fasta) { ch_fasta = file(params.fasta, checkIfExists: true) }
 
 /*
- * Check and print summary for parameters
+ * Check parameters
+ */
+Checks.aws_batch(workflow, params)     // Check AWS batch settings
+Checks.hostname(workflow, params, log) // Check the hostnames against configured profiles
+
+/*
+ * Print parameter summary
  */
-check_genome()                 // Check if genome exists in the config file
-check_hostname()               // Check the hostnames against configured profiles
-check_awsbatch()               // Check AWS batch settings
-summary = create_summary()     // Print parameter summary
+// Has the run name been specified by the user?
+// this has the bonus effect of catching both -name and --name
+run_name = params.name
+if (!(workflow.runName ==~ /[a-z]+_[a-z]+/)) {
+    run_name = workflow.runName
+}
```

Comment:
I think that with the upcoming parameter validation stuff, we can probably ditch this. Just use `workflow.runName` directly in the MultiQC process wrapper (and move this regex check there).

Reply:
I am still not clear as to what this is doing to be honest! What would we use [here](https://github.com/drpatelh/tools/blob/79622171cca6f321c5a268d9f2af297335f771eb/nf_core/pipeline-template/%7B%7Bcookiecutter.name_noslash%7D%7D/lib/JSON.groovy#L84)?